src/base-app.js

```diff
@@ -50,19 +50,20 @@ class FusionApp {
       // the type was never registered, throw error
       if (!registered.has(token)) {
         // Attempt to get default value
-        this.register(token, token());
+        const defaultValue = token();
+        if (defaultValue === undefined) {
```

Comment:
Explicit check on `undefined` to protect against the `.register` call below which special cases on [undefined](https://github.com/AlexMSmithCA/fusion-core/blob/b413573eb89930198e8e327334db1ebb38133d78/src/base-app.js#L12).

If we don't see a use case for having a default of `undefined`, I think throwing an error here is sufficient.